test/fdspaces.jl

```diff
@@ -645,3 +645,398 @@ end
+    end
+
+    function alloc_test_nested_expressions_5()
+        wvec = Geometry.WVector
+        If = Operators.InterpolateC2F(;
+            bottom = Operators.SetValue(0),
+            top = Operators.SetValue(0),
+        )
+        ∇c = Operators.DivergenceF2C()
+        @. cz =
+            cx * cy * ∇c(wvec(If(cy) * fx)) * ∇c(wvec(If(cy) * fx)) * cϕ * cψ # Compile first
+        p = @allocated begin
+            @. cz =
+                cx *
+                cy *
+                ∇c(wvec(If(cy) * fx)) *
+                ∇c(wvec(If(cy) * fx)) *
+                cϕ *
+                cψ
+        end
+        @test p == 2633232
+        @test_broken p == 0
+    end
+
+    function alloc_test_nested_expressions_6()
+        wvec = Geometry.WVector
+        Ic = Operators.InterpolateF2C()
+        ∇f = Operators.DivergenceC2F(;
+            bottom = Operators.SetValue(wvec(0)),
+            top = Operators.SetValue(wvec(0)),
+        )
+        @. fz =
+            fx * fy * ∇f(wvec(Ic(fy) * cx)) * ∇f(wvec(Ic(fy) * cx)) * fϕ * fψ # Compile first
+        p = @allocated begin
+            @. fz =
+                fx *
+                fy *
+                ∇f(wvec(Ic(fy) * cx)) *
+                ∇f(wvec(Ic(fy) * cx)) *
+                fϕ *
+                fψ
+        end
+        @test p == 2652320
+        @test_broken p == 0
+    end
+
+    function alloc_test_nested_expressions_7()
+        # similar to alloc_test_nested_expressions_8
+        wvec = Geometry.WVector
+        Ic = Operators.InterpolateF2C()
+        @. cz = cx * cy * Ic(fy) * Ic(fy) * cϕ * cψ # Compile first
+        p = @allocated begin
+            @. cz = cx * cy * Ic(fy) * Ic(fy) * cϕ * cψ
+        end
+        @test p == 0
+    end
+
+    function alloc_test_nested_expressions_8()
+        wvec = Geometry.WVector
+        Ic = Operators.InterpolateF2C()
+        @. cz = cx * cy * abs(Ic(fy)) * abs(Ic(fy)) * cϕ * cψ # Compile first
+        p = @allocated begin
+            @. cz = cx * cy * abs(Ic(fy)) * abs(Ic(fy)) * cϕ * cψ
+        end
+        @test p == 1589488
+        @test_broken p == 0
+    end
+
+    function alloc_test_nested_expressions_9()
+        wvec = Geometry.WVector
+        Ic = Operators.InterpolateF2C()
+        @. cz = Int(cx < cy) * abs(Ic(fy)) * abs(Ic(fy)) * cϕ * cψ # Compile first
+        p = @allocated begin
+            @. cz = Int(cx < cy) * abs(Ic(fy)) * abs(Ic(fy)) * cϕ * cψ
+        end
+        @test p == 2624
+        @test_broken p == 0
+    end
+
+    function alloc_test_nested_expressions_10()
+        wvec = Geometry.WVector
+        Ic = Operators.InterpolateF2C()
+        @. cz = ifelse(cx < cy, abs(Ic(fy)) * abs(Ic(fy)) * cϕ * cψ, 0) # Compile first
+        p = @allocated begin
+            @. cz = ifelse(cx < cy, abs(Ic(fy)) * abs(Ic(fy)) * cϕ * cψ, 0)
+        end
+        @test p == 1858176
+        @test_broken p == 0
+    end
+
+    println("##### Test allocs for nested expressions from real-world examples")
```

Comment:
Can we remove the print output, or comment it out?

Reply:
Done.